### PR TITLE
add cache control header support

### PR DESCRIPTION
**Pull request by jiechic**
添加一个cache注解，专门用来添加header的cache control属性
根据
https://developer.mozilla.org/zh-CN/docs/Web/HTTP/Headers/Cache-Control
把所有支持的客户端属性添加上，并使用other来扩展用户认为可以使用的自定义属性。

### CODE REVIEW

File: annotation/lib/http.dart [hunk 1]
```diff
@@ -271,3 +271,26 @@ class Part {
   final String? contentType;
   const Part({this.value, this.name, this.fileName, this.contentType});
 }
+
+@immutable
+class Cache {
```
Comment (NicolaVerbeeck) [inline annotation/lib/http.dart:276]: Better name this similar to the related header: `CacheControl`

File: generator/lib/src/generator.dart [hunk 2]
```diff
@@ -1271,9 +1278,51 @@ class RetrofitGenerator extends GeneratorForAnnotation<retrofit.RestApi> {
...
+
+    final cacheMap = _generateCache(m);
+    headers.addAll(cacheMap);
+
     return headers;
   }
 
+  Map<String, Expression> _generateCache(MethodElement m) {
+    final cache = _getCacheAnnotation(m);
+    var result=<String,Expression>{};
```
Comment (NicolaVerbeeck) [inline generator/lib/src/generator.dart:1290]: formatting

File: generator/lib/src/generator.dart [hunk 3]
```diff
@@ -1271,9 +1278,51 @@ class RetrofitGenerator extends GeneratorForAnnotation<retrofit.RestApi> {
...
+    if (cache != null && cache.toString() != '') {
+      var maxAge = cache.peek('maxAge')?.intValue;
```
Comment (NicolaVerbeeck) [inline generator/lib/src/generator.dart:1292]: use immutable variables where possible (final)

File: generator/lib/src/generator.dart [hunk 4]
```diff
@@ -1271,9 +1278,51 @@ class RetrofitGenerator extends GeneratorForAnnotation<retrofit.RestApi> {
...
+        maxStale != null ? 'max-stale=$maxStale' : '',
+        minFresh != null ? 'max-fresh=$minFresh' : '',
+        (noCache == true) ? 'no-cache' : '',
+        (noStore == true) ? 'no-store' : '',
+        (noTransform == true) ? 'no-transform' : '',
+        (onlyIfCached == true) ? 'only-if-cached' : '',
+        ...otherResult
+      ];
+
+      var value=values.where((element) => element != '').join(', ');
```
Comment (NicolaVerbeeck) [inline generator/lib/src/generator.dart:1319]: Formatting

**Review by NicolaVerbeeck — CHANGES_REQUESTED** [review]
See inline comments. Also add tests for the new functionality

**Comment by jiechic** [comment]
please check again

**Comment by NicolaVerbeeck** [comment]
Only 1 of the items has been addressed (the naming), other comments are still open (formatting, use final, ...)

**Comment by jiechic** [comment]
Sorry for not completing all changes in one submission
Please check the PR

File: generator/pubspec.yaml [hunk 5]
```diff
@@ -22,3 +22,8 @@ dependencies:
 dev_dependencies:
   test: 1.17.4
   source_gen_test: ^1.0.0
+
+dependency_overrides:
+  retrofit:
```
Comment (trevorwang) [inline generator/pubspec.yaml:27]: Please DO NOT commit these lines. It can break the package publishment.

**Review by trevorwang — CHANGES_REQUESTED** [review]
Please see the inline comment

File: generator/pubspec.yaml (hunk 5 above)
Comment (jiechic) [reply generator/pubspec.yaml]: I add cacheControl in annotation
the generator should be dependency new annotation version
i fix this commit but build will be faild

**Pull request merged**